graphbolt/src/cnumpy.h

```diff
@@ -0,0 +1,71 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file cnumpy.h
+ * @brief Numpy File Fetecher class.
+ */
+
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <torch/script.h>
+
+#ifdef __linux__
+#include <liburing.h>
+#include <unistd.h>
+#endif
+
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+namespace graphbolt {
+namespace storage {
+
+/**
+ * @brief Disk Numpy Fetecher class.
+ */
+class OnDiskNpyArray : public torch::CustomClassHolder {
+ public:
+  /** @brief Default constructor. */
+  OnDiskNpyArray() = default;
+
+  /** @brief Constructor with given file path. */
+  OnDiskNpyArray(std::string filename, torch::ScalarType dtype);
+
+  /** @brief Deconstructor. */
+  ~OnDiskNpyArray();
+
+  /**
+   * @brief Parse numpy meta data.
+   */
+  void ParseNumpyHeader();
+
+  /**
+   * @brief Read disk numpy file based on given index and transform to
+   * tensor.
+   */
+  torch::Tensor IndexSelectIOuring(torch::Tensor idx);
```

Comment:
If io_uring is guarded by `#if def __linux__`, then the `IndexSelectIOuring` should too.
I'd suggest that changing this to IndexSelect, and make IndexSelectIOuring a separate method, and dispatch in IndexSelect.

Reply:
nit: IndexSelectIOUring

Reply:
Are the IndexSelect and IndexSelectIOuring both internal methods? It looks like:
cnumpu.h
```
  torch::Tensor IndexSelect(torch::Tensor idx);
#ifdef __linux__
  torch::Tensor IndexSelectIOuring(torch::Tensor idx);
#endif  // __linux__
```

cnumpy.cc
```
torch::Tensor OnDiskNpyArray::IndexSelect(torch::Tensor idx) {
#ifdef __linux__
  return IndexSelectIOuring(idx);
#else
  return torch::empty({0});
#endif  // __linux__

#ifdef __linux__
torch::Tensor OnDiskNpyArray::IndexSelectIOuring(torch::Tensor idx) {
...
}
#endif  // __linux__
```